vmm/src/memory_manager.rs

```diff
@@ -1566,6 +1546,18 @@ impl MemoryManager {
     // pages touched during our bulk copy are tracked.
     pub fn start_memory_dirty_log(&self) -> std::result::Result<(), MigratableError> {
         for r in &self.guest_ram_mappings {
+            let user_addr = self
+                .guest_memory()
+                .memory()
+                .get_host_address(GuestAddress(r.gpa))
+                .unwrap();
+            self.vm
```

Comment:
This will fail for MSHV. right?

Reply:
Right. So will the `self.vm.get_dirty_log()` below. Of course, we expect both of them to work properly on `mshv` with your ongoing work in PR #2887.